src/OMSimulatorLib/SystemWC.cpp

```diff
@@ -163,6 +164,10 @@ oms_status_enu_t oms3::SystemWC::stepUntil(double stopTime)
       getModel()->emit(time);
   }
 
+  if(getParentSystem() && getParentSystem()->getType() == oms_system_tlm) {
+    reinterpret_cast<SystemTLM*>(getParentSystem())->disconnectFromSockets(this->getCref());
```

Comment:
Why isn't there a associated `connectToSockets` call?

Reply:
All submodels must be connected to sockets before they can begin intialization. Therefore the connect is handled in separate threads, started by SystemTLM. I guess we can use separate threads for disconnect as though, to avoid TLM stuff in  the SystemWC class.